### DISCUSSION TITLE
Artisan with 3 Phidget-based RTDs?

### DISCUSSION THREAD

**Discussion by N6GQ** [artisan-roaster-scope/artisan#749] [Q&A] [answered]
Newest artisan, set with ET and BT using 2 Phidget-based RTDs, the 3rd Phidget-based RTD is like to use for MET but I can’t find any way to do it. I’m using the TMP1200 modules and those work just fine for BT/ET but with “extra devices” I set up another TMP1200, but it just duplicates what the ET probe is reading, not the MET probe.

All are working with the Phidget control panel. How do I choose the 3rd RTD and display it? 

I’m going nuts here and and am the pointing of thinking a 3rd RTD just isn’t supported?

thanks,

jeff

**Comment by CarefreeBuzzBuzz** [comment]
See the quick start guide- use extra devices

Configure for Extra Devices Beyond TwoPermalink <https://artisan-scope.org/docs/setup/#configure-for-extra-devices-beyond-two>
Artisan supports two primary devices for BT and ET, and you can add additional LCDs and Curves by using Extra Devices tab on the Config» Devices dialog. Each physical device can have two channels, so on that tab you first select your device by its name in the drop down. Picking the correct device is critical, and if your device has more than two channels you might see some numbers after the name which indicates the additional channels such as Center 304 34. You pick a color for the display, apply a label/name for the device, add a formula to be applied to the device reading if any, and select if LCDs or Curves will be displayed. If you check the delta box, that will map the device to the right hand axis scale (the RoR axis) instead of the temperature axis.


----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Reply by N6GQ** [reply]
Ya, that’s what I tried to do - that seemed intuitively correct, however there is no 3rd RTD to choose - choosing the RTD that is there shows me the ET temp.

**Reply by poundy** [reply]
then it probably means you don't have the devices connected to your computer in a way that all sets of TMP1200's are visible.  Do you use a VINT hub of some kind?   Can you see all the devices properly in the Phidgets control software? 
Also, there's a few linked articles in https://artisan-scope.org/devices/phidgets/ that might be worth a read...

**Reply by N6GQ** [reply]
Yes, all devices show up in the phidgets control panel, I have 3 TMP1200's plugged into a VINT hub.

**Comment by CarefreeBuzzBuzz** [comment]
What you need is a symbolic assignment to read the correct channel of the extra device.

https://artisan-roasterscope.blogspot.com/2019/11/symbolic-formulas-basics-new-variables.html <https://artisan-roasterscope.blogspot.com/2019/11/symbolic-formulas-basics-new-variables.html>


----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Reply by N6GQ** [reply]
So artisan filled in Y3 in the extra devices UI under "Symb ET/BT" for ET Y(x). I think that's what you were suggesting needed to be done. 

I'm not trying to be dense, I just don't see any way to get that value into an LCD and curve.  If I select (checkbox) LCD and Curve in Extra Devices, it plots/shows the ET RTD temp, not the 3rd probe (which I'm calling MET). 

It seems to me there should be a way to choose each RTD/TMP1200 device. I have 3 TMP1200's plugged into ports 0,1 and 2 of the VINT Hub.

**Comment by CarefreeBuzzBuzz** [comment]
Did you explore the Extra device drop down and choose the one that has 23 on it for your choice?

Michael Herbert 
CarefreeBuzzBuzz

**Reply by N6GQ** [reply]
I don't have one with a 23 in the extra devices, see screenshot. This is a TMP1200 (I have 3 of them) for RTDs. The drop down is narrow so I can't see most of the item until its selected.
![Screenshot (2)](https://user-images.githubusercontent.com/56975772/144731095-f2da34d2-32d3-48e4-8e65-fd74ce867629.png)

**Comment by CarefreeBuzzBuzz** [comment]
Look for the VintHub not the TMP. 

Michael Herbert 
CarefreeBuzzBuzz

**Reply by N6GQ** [reply]
Ok, I see where you're going - so in Extra Devices I added "Phidget HUB IO 23" in addition to the Phidget TMP1200 that is already there, because the TMP1200+Symbolic ET/BT gives me ET in addition to the first TMP1200 that gives me BT. I was hoping that the "Phidget HUB IO 23" would then pass the temps for the 3rd TMP1200 through, but alas it does not seem to. 

Do I need another symbolic mapping for that HUB IO device? 

Sorry, I'm just not understanding. Doesn't seem like it should be this difficult - I would have guessed that there would just be a device for each TMP1200, add as many as you want, connect them with known variables like ET/BT, and the rest be able to plot/display as desired. Maybe I'm making it too hard.

**Comment by N6GQ** [comment, marked as answer]
Ok, I finally figured it out.

I added a 2nd Extra Device (TMP1200) and put Y5 in the y1(x) box. 

I think I have a grip on how this works now. 

Thanks for the guidance...
